WalletWasabi/Wallets/BlockDownloadStats.cs

```diff
@@ -0,0 +1,153 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using NBitcoin.Protocol;
+using WalletWasabi.Crypto.Randomness;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Wallets;
+
+public class BlockDownloadStats
+{
+	private double AverageLastSample { get; set; }
+	private double StandardDeviationLastSample { get; set; }
+	private int ConsecutiveFails { get; set; }
+	private IEnumerable<BlockDl> Sample { get; set; } = Enumerable.Empty<BlockDl>();
+	private List<BlockDl> History { get; } = new();
+	private object Lock { get; } = new();
+	
+	public void AddBlockDl(IPAddress from, double msElapsed, bool success)
+	{
+		lock (Lock)
+		{
+			History.Add((new BlockDl(from, msElapsed, success)));
+			Sample = History.Skip(Math.Max(0, History.Count - Const.SampleSize));
+			(AverageLastSample, StandardDeviationLastSample) = 
+				GetAverageStandardDerivation(Sample.Where(x => x.Success));
+			ConsecutiveFails = success ? 0 : ConsecutiveFails + 1;
+		}
+	}
+
+	private static (double, double) GetAverageStandardDerivation(IEnumerable<BlockDl> collection)
+	{
+		return MathUtils.AverageStandardDeviation(collection.Select(x => x.MsElapsed));
+	}
+	
+	public int NodeTimeoutStrategy(int lastTimeout)
+	{
+		lock (Lock)
+		{
+			int baseTimeout;
+			if (AverageLastSample > 0 && Sample.Count(x => x.Success) > Const.SampleMinSuccesses)
+			{
+				// Mean time spent recently to download a block including the spread
+				baseTimeout = (int)Math.Round(AverageLastSample + Math.Max(Const.MinimumSpread, StandardDeviationLastSample * Const.StandardDerivationWeight));
+			}
+			else
+			{
+				// Same but for History if we don't have enough info in sample.
+				var historyAvgSd = GetAverageStandardDerivation(History.Where(x => x.Success));
+				if (historyAvgSd.Item1 > 0)
```

Comment:
Try to avoid Item1, use names tuples like above 
(AverageLastSample, StandardDeviationLastSample) = GetAverageStandardDerivation

It helps readability.